modules/RTC/DataChannels.js

```diff
@@ -247,4 +239,37 @@ DataChannels.prototype._some = function (callback, thisArg) {
     }
 };
 
+/**
+ * Sends passed object via the first found open datachannel
+ * @param jsonObject {object} the object that will be sent
+ */
+DataChannels.prototype.send = function (jsonObject) {
```

Comment:
I am referring to the situation in which the data channel has not opened
(for example because of the jvb bug), in which case calls to
DataChannels.send silently fail (that is, do not result in a message being
sent).

Boris

On Jul 22, 2016 17:36, "hristoterezov" <notifications@github.com> wrote:

> In modules/RTC/DataChannels.js
> <https://github.com/jitsi/lib-jitsi-meet/pull/181#discussion_r71896953>:
>
> > @@ -247,4 +239,37 @@ DataChannels.prototype._some = function (callback, thisArg) {
> >      }
> >  };
> >
> > +/**
> > + * Sends passed object via the first found open datachannel
> > + * @param jsonObject {object} the object that will be sent
> > + */
> > +DataChannels.prototype.send = function (jsonObject) {
>
> @bgrozev <https://github.com/bgrozev> But we are checking if the channel
> is open before sending. What is the use case you are referring to?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jitsi/lib-jitsi-meet/pull/181/files/dd24d1bee1245c50ff41e33f63bb4ffccc7897fb#r71896953>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADHQuw-QJPHZk7r52XpUQGKFnr2lLFI3ks5qYODHgaJpZM4JRKgM>
> .
>